image_writers_guide/sti.adoc

```diff
@@ -10,70 +10,52 @@
-environment variable from the builder image (default location is `/tmp`). For this
-to happen your image must supply tar archiving utility (command `tar` available in `$PATH`)
-and command line interpreter (command `/bin/sh`). Doing so will allow your image to
-use the fastest possible build path, because in all other cases when either
-`tar` or `/bin/sh` is not available, sti build will be forced to perform an additional
-docker build to put both sources and scripts inside the image and only then run the
-usual sti build procedure (sti will do this automatically if tar and /bin/sh are not found).
-See the following diagram for how build workflow looks like:
+See the following diagram for the basic STI build workflow:
 
-image:sti-flow.png[STI workflow]
+.Build workflow
+image::sti-flow.png[STI workflow]
 
-\* Run build's responsibility is to untar the sources, scripts and artifacts (if such
-exist) and invoke `assemble` script. If this is second run (after catching `tar`/`/bin/sh`
-error) it's responsible only for invoking `assemble` script, since both scripts and
-sources are already there.
+////
+* Run build's responsibility is to untar the sources, scripts and artifacts (if such exist) and invoke `assemble` script. If this is second run (after catching `tar`/`/bin/sh` error) it's responsible only for invoking `assemble` script, since both scripts and sources are already there.
+////
 
+== STI Scripts
+STI expects you to supply the following scripts:
 
-== STI scripts
+* link:#assemble[`assemble`] (required)
+* link:#run[`run`] (required)
+* link:#save-artifacts[`save-artifacts`] (optional)
+* link:#usage[`usage`] (optional)
 
-STI expects you, as the builder image author, to supply the following scripts:
+All of the scripts can be written in any programming language, as long as it is executable inside the builder image.
 
-* required:
-    * link:[#assemble]
-    * link:[#run]
-* optional:
-    * link:[#save-artifacts]
-    * link:[#usage]
+NOTE: The examples in the following sections are written in Bash and it is assumed all tar contents are unpacked into the [filename]#/tmp/sti# directory.
 
-All of the scripts can be written in any programming language, as long as it is
-executable inside the builder image.
+[[assemble]]
+*`assemble`*
 
-=== assemble
+The `assemble` script is responsible for building the application artifacts from source and placing them into appropriate directories inside the image. The workflow for the `assemble` script is:
 
-The `assemble` script is responsible for building the application artifacts from source,
-and place them into appropriate directories inside the image. The workflow for `assemble` is:
+. Restore build artifacts. If you want to support incremental builds, make sure to define link:#save-artifacts[`save-artifacts`] as well.
+. Place the application source in the desired location.
+. Build the application artifacts.
+. Install the artifacts into locations appropriate for running.
```

Comment:
[Suggestion]
Install the artifacts into locations that are appropriate for running the scripts.

I wasn't sure what we're referring to with "appropriate for running". I'm assuming we mean running the scripts themselves?